dask_kubernetes/core.py

```diff
@@ -440,15 +482,16 @@ def _namespace_default():
     return 'default'
 
 
-def select_workers_to_close(s, n):
-    """ Select n workers to close from scheduler s """
-    assert n <= len(s.workers)
+def select_workers_to_close(scheduler, n_to_close):
+    """ Select n workers to close from scheduler """
+    workers = list(scheduler.workers.values())
+    assert n_to_close <= len(workers)
     key = lambda ws: ws.info['memory']
-    to_close = set(sorted(s.idle, key=key)[:n])
+    to_close = set(sorted(scheduler.idle, key=key)[:n_to_close])
 
-    if len(to_close) < n:
-        rest = sorted(s.workers.values(), key=key, reverse=True)
-        while len(to_close) < n:
+    if len(to_close) < n_to_close:
+        rest = sorted(workers, key=key, reverse=True)
+        while len(to_close) < n_to_close:
```

Comment:
Eventually we might want to leverage `Scheduler.workers_to_close`. It won't take into account succeeded pods but may be useful just to centralize logic.  

This doesn't need to be now though, I just wanted to increase awareness that this logic also existed in the core library.

Reply:
Yes I saw that and was wondering why you did not use it in the first place. I wanted this PR to only fix the original problem but indeed it might be a good idea to centralize logics in a following PR.